### ISSUE TITLE
Accessiblity WCAG 2.1

### ISSUE THREAD

**Issue by harryhaigh** [primefaces/primereact#1887]
I am just trying out PrimeReact and noticed a lot of the component do not meet the WCAG 2.1 standards. 
An example is the Tabs where it is not keyboard accessible and it is not a 1 tab stop. 

Are there any plans to make this WCAG 2.1 compliant?

**Comment by MarcusMorba** [comment]
@melloware & @mertsincan - any updates to the WCAG 2.1 issues? When can we expect them to be solved?

**Comment by melloware** [comment]
@MarcusMorba I don't work for PrimeTek but the update I got was PrimeTek was working on "PrimeAG -> Prime Accessibility Guide. It is a spec that will be implemented by all Prime libraries."

After the guide is created then I assume work in PrimeReact will begin to implement that spec! At least hopefully!

**Comment by MarcusMorba** [comment]
Found an update here: https://www.primefaces.org/primereact/accessibility/

Accessibility is a majorn concern of the Prime UI libraries and PrimeReact is no exception. [PrimeTek](https://www.primetek.com.tr/) teams have initiated a significant process to review and enhance the accessibility features of the components. This guide documents the foundation of the general guidelines that PrimeReact will follow and each component documentation will have a separate Accessibility section that states the keyboard support, screen reader compatibility, the implementation details along with tips to achieve WCAG compliancy. 

This work has been initiated in Q2 2022 and planned to be completed by early Q3. PrimeReact will be the reference implementation which then will be ported to PrimeFaces, PrimeNG and PrimeVue.

**Comment by melloware** [comment]
Looks like they are working on it for PrimeReact first.  If you look through the commits you can see they are going component by component addressing `a11y` standards.

**Comment by Luisa-Cavalheiro** [comment]
I understand implementing a11y is kind of ant's work. However, I'd love to follow the improvements in this subject, and eventhough I'm a begginer in software development, I'd love to take part in this project somehow.

Thanks

**Comment by melloware** [comment]
@Luisa-Cavalheiro its an open an welcoming community here!   I do recommend checking out the PrimeVue showcase https://primevue.org/installation  If those components have the right ARIA/A11Y stuff then you know that is coming to PrimeReact in 10.0!

**Comment by MarcusMorba** [comment]
Thanks for the update. Guess it´s time to close this, even if it´s still work in progress - not all components have an a11y section yet. Documentation: https://primereact.org/accessibility/ and on every component page in the "accessibility"-section (if available).

**Comment by melloware** [comment]
@MarcusMorba going to leave this open until all of PrimeReact is a11y aligned with PrimeVue.

**Comment by MarcusMorba** [comment]
@melloware - then maybe WCAG 2.1 should be changed to WCAG 2.2 as this will be the next upcoming standard soon. 
"WCAG 2.2 is scheduled to be completed and published in 2023 Q3 (July, August, September)." (https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/)

**Comment by melloware** [comment]
Updated title!

**Comment by Doblanko** [comment]
What is the estimated time of completion? Accessibility is critical for our application and we need to decide how to address these issues. Many components are fundamentally not accessible at the moment.

From [this thread](https://github.com/primefaces/primereact/issues/1413) it appears that the fixes got pushed from version 9 to version 10, and now 11.

**Comment by melloware** [comment]
Its 11.0.0 now.  If you check out PrimeVue showcase which has accessibility addressed anything in PrimeVue is going to be ported to PrimeReact for 11.0.0.

**Comment by Doblanko** [comment]
When is the estimated release of 11.0.0?

**Comment by melloware** [comment]
> When is the estimated release of 11.0.0?

I don't work for PrimeTek but I suggest if you need this urgently your company look into PRO support: https://primereact.org/support/

**Comment by melloware** [comment]
A lot of accessibility has been added to 10.5.3 if anyone wants to test and report issues.

**Comment by MarcusMorba** [comment]
@melloware - hope you will find some people for this. As my focus got back to Drupal at the moment and other FOSS web-component libraries like e.g. KoliBri (https://github.com/public-ui/kolibri) I no longer have resources to contribute here. Sorry.
